examples/ACE/utils/NNIAP.jl

```diff
@@ -72,12 +130,45 @@ function learn!(nniap, ds, opt::Flux.Optimise.AbstractOptimiser, epochs, loss, w
         grads = ∇loss(nniap.nn, nniap.iap, ds, w_e, w_f)
         Flux.update!(optim, nniap.nn, grads[1])
         # Logging
-        curr_loss = loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        loss_time = @elapsed curr_loss = loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        println("loss_time", loss_time)
         push!(losses, curr_loss)
         println(curr_loss)
     end
 end
 
+function learn!(nace, ds_train, opt, n_epochs, n_batches, loss, w_e, w_f, _device)
+    nn = nace.nn |> _device
+    iap = nace.iap |> _device
+    ds = ds_train |> _device
```

Comment:
This does not transfer the dataset to the GPU. Furthermore, the function `learn!` will transfer batches of training data to the GPU. For large data sets it is not possible to transfer all the training data to the GPU at once, which is one of the reasons why batches are used. Please, remove `|> _device`